### PR TITLE
Add protobuf messages

### PR DESCRIPTION
**Pull request by djpetti**
I'm still not entirely clear on whether the participants actually have to listen on a port as opposed to just connecting to an open port the coordinator and receiving messages that way. If so, this will need to be changed.

### CODE REVIEW

File: project3/proto/pub_sub_messages.proto [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  /// The ID of the participant that this message originated from.
+  uint32 origin_id = 2;
+}
+
+/**
+ * @brief Encapsulates all messages that can be received by the coordinator. This is
+ * so we can simply parse this message type from the network and worry later about
+ * the actual message type.
+ */
+message CoordinatorMessage {
```
Comment (Nickster258) [inline project3/proto/pub_sub_messages.proto:49]: My only concern with this is that it encapsulates messages that the coordinator will be receiving, but the name implies it contains messages the coordinator will be sending. Should this be instead generalized for both, or possibly renamed to reflect project 1 and 2 structure?

**Review by Nickster258 — COMMENTED** [review]
All in all it looks like these messages are all we would need for having a multicast server and client communicate. I only have one concern about the message encapsulation.

File: project3/proto/pub_sub_messages.proto (hunk 1 above)
Comment (djpetti) [reply project3/proto/pub_sub_messages.proto:49]: I actually don't like the project1/2 structure as much. In practice, you really only need to do this encapsulation for messages that are being received, not send, hence I think it's better to name these encapsulation messages after the receiving end. I know that would certainly make more sense to me if I was skimming the coordinator code and came across a message of this type.

Comment (Nickster258) [reply project3/proto/pub_sub_messages.proto:49]: Yeah, that does actually make more sense in that case.

**Review by Nickster258 — APPROVED** [review]

**Pull request merged**